src/log/iobuf.rs

```diff
@@ -617,3 +797,15 @@ fn bump_offset(v: u32, by: u32) -> u32 {
     assert_eq!(by >> 24, 0);
     v + by
 }
+
+#[inline(always)]
+fn debug_delay() {
+    #[cfg(test)]
+    {
+        use rand::{Rng, thread_rng};
+
+        if thread_rng().gen_weighted_bool(1000) {
+            std::thread::sleep(std::time::Duration::from_millis(100));
+        }
+    }
+}
```

Comment:
This thing (which gets fully elliminated from non-test code due to the `inline(always)` and `cfg(test)`) is a nice interleaving exerciser, ensuring we get great coverage for the lock-free algorithm in this system.